### PR TITLE
[RGL] Upgrade to v11 + Mesh2PCD + MeshSharing 

### PR DESCRIPTION
**Pull request by prybicki**

### CODE REVIEW

**Review by mackierx111 — COMMENTED** [review]
RGL's node design is great👍 

I made several comments. 
Also, dll loading fails in windows.
image

**Comment by msz-rai** [comment]
@mackierx111
Thank you for the review.
To help solve the problem with dll please provide your windows version, GPU driver version, and CUDA version.

**Comment by mackierx111** [comment]
@msz-rai 
I updated the Nvidia drivers to the latest and was able to run it on windows. Thank you!

File: Assets/AWSIM/Scenes/Main/PointCloudMapping/Scripts/Mapper/PointCloudMapper.cs [hunk 1]
```diff
@@ -39,45 +38,58 @@ public class PointCloudMapper : MonoBehaviour
...
+
+        public string GetOutputPcdFilePath()
+        {
+            return $"{Application.dataPath}/{outputPcdFilePath}";
+        }
 
         public void Start()
         {
-            mappingSensors = new List<IMappingSensor>(vehicleGameObject.GetComponentsInChildren<IMappingSensor>());
+            mappingSensors = new List<RGLMappingAdapter>(vehicleGameObject.GetComponentsInChildren<RGLMappingAdapter>());
```
[the remark this answers is not in the document]
Comment (msz-rai) [reply Assets/AWSIM/Scenes/Main/PointCloudMapping/Scripts/Mapper/PointCloudMapper.cs:56]: Thank you for the suggestion. I agree with you, but if we are planning to support multi-sensor mapping, components are necessary for easy configuration. I'd leave it as it is to simplify changes in the future.

File: Assets/AWSIM/Scenes/Main/PointCloudMapping/Scripts/Mapper/RGLMappingAdapter.cs [hunk 2]
```diff
@@ -1,31 +1,49 @@
...
     {
         [SerializeField]
         [Tooltip("Resolution to sub-sample point cloud data. Set leaf size to 0 if you don't want to sub-sample.")]
         private float leafSize;
-        
+
+        private Vector3 worldOriginROS;
+        private string outputPCDFilePath = "output.pcd";
+
+        public PointCloudMapper mapper;
```
[the remark this answers is not in the document]
Comment (msz-rai) [reply Assets/AWSIM/Scenes/Main/PointCloudMapping/Scripts/Mapper/RGLMappingAdapter.cs]: Thank you. I have applied your suggestion

File: Assets/RGLUnityPlugin/Scripts/LidarSensor.cs [hunk 3]
```diff
@@ -152,29 +148,41 @@ public void FixedUpdate()
...
+            // Set lidar pose
+            Matrix4x4 lidarPose = gameObject.transform.localToWorldMatrix;
+            rglGraphLidar.UpdateNodeRaysTransform(lidarPoseNodeId, lidarPose);
+            rglSubgraphToLidarFrame.UpdateNodePointsTransform(toLidarFrameNodeId, lidarPose.inverse);
+
+            rglGraphLidar.Run();
 
-            return outputData;
+            // Could be moved to PointCloudVisualization
+            if (GetComponent<PointCloudVisualization>().isActiveAndEnabled == true)
```
Comment (pijaro) [inline Assets/RGLUnityPlugin/Scripts/LidarSensor.cs:180]: nitpick
```suggestion
            if (GetComponent<PointCloudVisualization>().isActiveAndEnabled)
```

File: Assets/AWSIM/Scenes/Main/PointCloudMapping/Scripts/Mapper/PointCloudMapper.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mackierx111) [reply Assets/AWSIM/Scenes/Main/PointCloudMapping/Scripts/Mapper/PointCloudMapper.cs:46]: How about implementing it as a single member in the current situation and changing it to an array when multi-sensor support is provided?

[the remark this answers is not in the document]
Comment (msz-rai) [reply Assets/AWSIM/Scenes/Main/PointCloudMapping/Scripts/Mapper/PointCloudMapper.cs]: Good point! Thank you

**Review by mackierx111 — APPROVED** [review]
Thanks for the implementation!

**Pull request merged**